cluster/log-dump/log-dump.sh

```diff
@@ -141,7 +141,7 @@ function save-logs() {
       fi
     else
       case "${KUBERNETES_PROVIDER}" in
-        gce|gke)
+        gce|gke|kubernetes-anywhere)
```

Comment:
Tote that kubernetes anywhere tests are pretending to be a 'gce' provider when calling this log logic. Depending on what you are trying to do, you may need to change that (or add to it).

https://github.com/kubernetes/test-infra/blob/master/kubetest/anywhere.go#L273

Reply:
@jessicaochen yep, am fixing that in the companion PR https://github.com/kubernetes/test-infra/pull/5164